Syndicate Bundle Overhaul

Kierany9 opened the pull request:
Syndicate Bundles, a traitor item that selects a bunch of items specialized items worth just over 20 TC, have been rather bad for a long time, often discarded for the far superior Syndicate Crate, which would grant you 50 TC worth of random items. Combine this with balancing changes which reduced the TC cost of a lot of utility items that came with the bundles and you have a recipe for a shitty item that may not even give you your money's worth. No more! Syndicate bundles have been reworked, many of which carry many unique items and allowing for more gimmicky but nonetheless powerful playstyles.

this stuff may need nerfing or other balancing please tell me if it does

**Spy Bundle:**
- 4x DNA Randomizers
- 3x Agent ID Cards
- 3x Syndicate Encryption Keys
- 2x Greater Mindslave Implants
- Syndicate Surgery Dufflebag
- Energy Dagger

The Spy Bundle is focused on obtaining a small gang of henchmen to help you complete your objectives. While you have plenty of items to help team communication, you get nothing to arm them or to help you capture them, only a mere edagger to defend yourself with. Resourcefullness is the key to getting the most out of this kit.

**Snake Bundle:**
- Sketchin Pistol (w/ Supressor and spare mag)
- Combat Knife
- CQC Gloves
- Dart Gun
- 5x Bioterror Syringes
- 2x C4
- Agent ID Card
- Syndicate Cigarretes

Fufill your fantasies of becoming Big Boss today with the Snake Bundle. This bundle is perfectly equipped for 1v1 combat and is geared at fighting and infiltrating Security/Command rather than murderboning civilians.

**Operative Bundle:**
- Blood Red Hardsuit
- Sketchin Pistol (w/ 3 spare mags)
- Syndicate Minibomb
- Detomatrix Cartrige
- Syndicate Encryption Key
- Powersink Beacon
- Singularity Beacon
- Syndicate Bomb Beacon

This bundle is for amaeteur Nuke Ops who want to spread panic throughout the station. And that's it really.

**Matrix Bundle:**
- Revolver (w/ spare mag)
- 2x Sketchin Pistol (w/ 8 spare mags)
- 2x C4
- Leather Overcoat
- Sunglasses

The Matrix Bundle offers one way to complete your objectives and one way only. That way involves explosions, gunshots and trenchcoats galore. Dual wield your Sketchins for ultimate badassery.

**Implant Bundle:**
- Freedom Implant
- Adrenal Implant
- Storage Implant
- EMP Implant
- Macrobomb Implant
- Mindslave Implant

Possibly one of the strongest bundles in the game, simply because it comes with an uplink implant out of all things. You get 10 TC to buy whatever you wish as well as the fuckton of implants you already have including mindslave and **macrobomb**.

**Hacker Bundle:**

- Hacked AI Law Module
- Agent ID Card
- Chameleon Jumpsuit
- Armoured Labcoat
- Binary Encryption Key
- Syndicate Encryption Key
- Emag
- AI Detector
- Camera Bug
- AI Toy (Disguised Law Module)
- C4

The Hacker Bundle now actually works like a hacker's bundle. Subverting the AI is nolonger his only trick, he now has a unique Agent ID Card with access to Tcomms and RnD, allowing him to sabotage the radio with potent scripts or steal prototype weapons from RnD(before wiping the servers like a jackass).

**Dark Lord Bundle:**
- 2x Energy Sword
- Summon Item Spellbook
- Adrenal Implant
- Telekineses SE Injector
- Mindslave Implant
- Agent ID Card
- Dark Lord Robes

The Dark Lord Implant has received minor buffs with a mindslave implant(to turn people to the dark side of course), adrenals for stun resistance and the Summon Item spellbook to recover the double esword incase you somehow manage to lose it.

**Ninja Bundle:**
- Energy Sword
- Cloak of Darkness SE Injector
- Throwing Star Belt
- 2x Throwing Star Boxes
- Pickpocket Gloves
- Smoke Spellbook
- 2x Airlock Charge
- Soap
- Balaclava

While somewhat similar to the Dark Lord, the Ninja Bundle is more focused on maintinence prowling and otherwise staying out of sight, keeping pursuers at bay with smokescreens and an abundance of throwing stars.

**Explosives Bundle:**
- 20x C4
- 4x Syndicate Minibombs
- 2x Syndicate Bombs

BOOM BOOM BOOM BABY!

Kierany9 commented:
shit i forgot logs hang on

JohnGinnane commented on code/game/objects/items/weapons/storage/belt.dm:
```diff
@@ -150,6 +150,32 @@
+	new /obj/item/weapon/throwing_star(src)
+	new /obj/item/weapon/throwing_star(src)
+	new /obj/item/weapon/throwing_star(src)
+	new /obj/item/weapon/throwing_star(src)
```
Can I suggest using a loop? It would allow you to easily change the number of stars.

As an addition, I'm not sure if possible but you could have a parameter (default value 15) which is how many stars to spawn?

JohnGinnane commented on code/game/objects/items/weapons/storage/uplink_kits.dm:
```diff
@@ -2,100 +2,175 @@
+			new /obj/item/weapon/c4(src)
+			new /obj/item/weapon/c4(src)
+			new /obj/item/weapon/c4(src)
+			new /obj/item/weapon/c4(src)
```
Please use a loop for this :(

TDcoolguy300 replied:
Loop would be nice

Kierany9 replied:
No other item containers use loops, and from what I've tested, they don't work for whatever reason unless I'm extremely dumb.

Which I probably am.

JohnGinnane replied:
https://github.com/HippieStationCode/HippieStation13/blob/5462010e5bcc8a61ad5618ffd0f04f07eeb570b0/code/game/objects/structures/crates_lockers/closets/secure/freezer.dm#L10

Here's a perfect example!

Kierany9 replied:
Why coding at midnight is a bad idea

JohnGinnane commented:
Code looks good, though it would be nice if you replaced anywhere that has 5 of the same line with a loop for ease-of-maintenance

Kierany9 commented:
- Removed Instant Summon from the Ninja bundle and replaced with two airlock charges and one bar of soap to differentiate it from the Dark Lord bundle and improve the stealth and trap-laying aspects of the bundle.

- Removed Voice Changer and Encryption Key from the Snake bundle which is already good enough without it.

- Added loops for the C4 and Ninja bundles

JohnGinnane commented:
Hooray! Code looks good. I think this might need to be discussed for balance purposes though?

Kierany9 commented:
Definitely, we need a proper discussion on this before getting it merged.

carlarc commented:
for example: why only 49 c4? my inner ocd is itching

Nokington commented:
I like the idea but it seems like a bit much

blobbernaut commented:
my proper discussion is this will actually make me bother purchasing bundles, whereas now, I never do.

Lawligagger commented:
I'm for it

Nebyab commented:
yeah alright, seems mostly fine. C4 one might be a bit shitty, but let's a live a little and see what happens.

Kierany9 commented:
49 c4s fill exactly seven rows of the box they come in. Having one extra c4 on the eighth row was itching my own inner ocd.

Crystalwarrior commented:
I absolutely love every one of these, especially the snake one.
Though can you add a portable solid box item (deploy and retrieve big box)? Means that you can easily infiltrate into sec with the box without having to abandon your favorite box.

JohnGinnane commented:
That's an amazing idea!

On Sat, Apr 30, 2016 at 7:09 PM, Crystalwarrior <notifications@github.com>
wrote:

> I absolutely love every one of these, especially the snake one.
> Though can you add a portable solid box item (deploy and retrieve big
> box)? Means that you can easily infiltrate into sec with the box without
> having to abandon your favorite box.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/HippieStationCode/HippieStation13/pull/2233#issuecomment-215984460>
>

Carbonhell commented:
doesn't implant bundle already exist?
btw i fucking love the idea god bless you magnificent bastard

Kierany9 commented:
A lot of these bundles already exist, I just added buffs, nerfs, changes or total reworks in the case of the spy bundle.

McBawbaggings commented:
Spy bundle looks dope as fuck.
Snake bundle is cool.
Matrix bundle could use some sunglasses.
Nukie bundle is cool.
Hacker bundle ID card seems silly. You have an emag which removes tcomm restrictions and RnD restictions (I think). It should be about hacking in rather than waltzing in. You can also have your subverted AI do it for you too.
Implant bundle seems a bit nuts, but why not?
Dark lord bundle is cool.
Ninja bundle is interesting in.
Explosives bundle is a bit memey, it should be an assortment of explosives rather than more C4 than your body has room for.

Not quite looked at the TC cost for these items but none of these should be making the random item crate totally worthless since there's an element of randomness.

I'll let you know what I think once I've had a look.

dreamcrab commented:
i like a lot of these, i could probably come up with a few as well. more bundles is fun, if we had a lot of them of varying power it could be a lot of fun

Kierany9 commented:
- Improved Explosive Bundle
- Added shades to Matrix Bundle
- Removed Uplink Implant from the Implant Bundle
- Removed the Hacker ID card from the Hacker Bundle, replaced with an Agent ID card, an armoured labcoat and a chameleon jumpsuit.

JohnGinnane commented on code/game/objects/items/weapons/storage/uplink_kits.dm:
```diff
@@ -2,100 +2,134 @@
 
 /obj/item/weapon/storage/box/syndicate/New()
```
I suggest moving the repeated "return" after the switch statement, instead of in each "case"

JohnGinnane commented:
Code looks good

Kierany9 commented:
merge conflicts

send help

Lawligagger commented:
http://i.imgur.com/Una2rBn.png

Kierany9 commented:
Closing, can't be arsed and am _way_ too busy at the moment to solve conflicts.

carlarc commented:
asshole

Crystalwarrior commented:
bitch fuck mcgee
